src/main/java/net/mcreator/element/types/VillagerTrade.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.parts.VillagerProfession;
+import net.mcreator.workspace.elements.ModElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VillagerTrade extends GeneratableElement {
+
+	public List<CustomTradeEntry> tradeEntries;
+
+	private VillagerTrade() {
+		this(null);
+	}
+
+	public VillagerTrade(ModElement element) {
+		super(element);
+		tradeEntries = new ArrayList<>();
+	}
+
+	public static class CustomTradeEntry {
+
+		public VillagerProfession villagerProfession;
+		public List<Entry> entries;
+
+		public static class Entry {
+
+			public MItemBlock price1;
+			public int countPrice1;
+
+			public MItemBlock price2;
+			public int countPrice2;
+
+			public MItemBlock sale;
+			public int countSale;
+
+			public int level;
+			public int maxTrades;
+			public int xp;
+
+			public double priceMultiplier;
+
+			public Entry() {
```

Comment:
What is the purpose of this constructor?

Reply:
It's used in `JVillagerTradeEntry`, that's what allows all the fields to make one single entry.

Reply:
But there, you set different values, why do you set values here?

Reply:
I'm setting default values, so when someone creates a trade, he has mostly common values (for priceMultiplier, it's almost always 0.05 for example)

Reply:
But these values are set in UI elements, what is the purpose of them here?

Reply:
Empty public constructor is provided to each class by default. No need to keep this constructor if it doesn't do anything specific.

Reply:
The `Entry()` constructor is used in tests and other files

Reply:
But this exists in Java by default, why explicit definition?

Reply:
Done, empty constructor removed

Reply:
You can mark resolved @KlemenDEV